contracts/core/EntryPoint.sol

```diff
@@ -56,6 +59,14 @@ contract EntryPoint is IEntryPoint, StakeManager {
         try this.innerHandleOp(userOp.callData, opInfo, context) returns (uint256 _actualGasCost) {
             collected = _actualGasCost;
         } catch {
+            bytes32 innerRevertCode;
+            assembly {
+                returndatacopy(0, 0, 32)
+                innerRevertCode := mload(0)
+            }
+            // handleOps was called with gas limit too low. abort entire bundle.
+            require(innerRevertCode != INNER_OUT_OF_GAS, "AA95 out of gas");
```

Comment:
A malicious paymaster could revert with 0xdeaddead from its postOp.  It would revert the entire bundle without a FailedOp, making it impossible to know what op caused the revert, and which paymaster to ban.

Since the op itself can't cause this OOG situation (both the account and the paymaster's postOp are called with gas limits), then it's either caused by the bundler, or by a malicious paymaster.  Therefore, why not revert it with a FailedOp, reporting the paymaster?  If the bundler is malicious then paymaster reputation held by it, is irrelevant.  And if the bundler is not malicious than it can only be the paymaster's fault.  No way for the account to cause this, right?

Reply:
changing to FailedOp
(and it can only be with a staked paymaster, since it requires a "postOp" )

Reply:
Actually, I can't find where the EIP says postOp requires staking.  I only see the storage part.